OffchainLabs/arb-token-bridge
# fix: properly add and remove token lists

### Summary

Token were previously attached to only one token list, when the token list was removed, all token attached to this list were removed, even if those tokens were part of another list. Tokens now have a `set` containing the different token lists attached to it.

### Steps to test

- Impersonate `0xc5ed2333f8a2c351fca35e5ebadb2a82f5d254c3` address in Frame/Impersonator
- Open token lists
- Click on “Manage token lists”
- Uncheck `Arbitrum Whitelist Era`
- Check `Arbitrum Whitelist Era`
- Uncheck `Arbitrum Whitelist Era`
- Go back

Without the fix, only `ETH` should be displayed. With the fix, token present in Uniswap list should still be displayed (for example `SPELL`, `USDT`). `MAGIC` should not be displayed anymore
